File: natalis-backend-main/src/main/java/dev/ved/natalis/test_service/dto/DashboardStatsUpdateEvent.java
package dev.ved.natalis.test_service.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class DashboardStatsUpdateEvent {

    private long totalTestsIncrement;
    private long testsTodayIncrement;
    private long testsThisMonthIncrement;
}

File: natalis-mobile-app-main/lib/models/dashboard_stats_update_event.dart
class DashboardStatsUpdateEvent {
  final int totalTestsIncrement;
  final int testsTodayIncrement;
  final int testsThisMonthIncrement;

  DashboardStatsUpdateEvent({
    required this.totalTestsIncrement,
    required this.testsTodayIncrement,
    required this.testsThisMonthIncrement,
  });

  factory DashboardStatsUpdateEvent.fromJson(Map<String, dynamic> json) {
    return DashboardStatsUpdateEvent(
      totalTestsIncrement: json['totalTestsIncrement'] ?? 0,
      testsTodayIncrement: json['testsTodayIncrement'] ?? 0,
      testsThisMonthIncrement: json['testsThisMonthIncrement'] ?? 0,
    );
  }
}
